### DISCUSSION TITLE
How do I use the capture plug-in？

### DISCUSSION THREAD

**Discussion by Mr-LZG** [FreeRDP/FreeRDP#9773] [Q&A]
I used the capture plug-in, but it only sent session information messages and session end messages, and did not send graphics data. is always returned in the following code 
![image](https://github.com/FreeRDP/FreeRDP/assets/22317592/9977ac7f-f551-4aa5-8fc3-f8edefafa05a)

Please tell me how to use the capture plug-in correctly.

**Comment by akallabeth** [comment]
and you are using which version?
if it is 2.x then the plugin requires the appropriate gfx settings enabled.
if it is 3.x then the plugin does no longer works for that purpose as we do not decode graphics data on proxy.

[note] we added a generic session dump mechanism which requires setting `FreeRDP_TransportDump` for a connection. With that option it is possible to write the whole session raw data unencrypted to disk.
This is currently not exposed by the proxy binary.

The dump could then be replayed (wouldrequire a custom `player` that emulates RDP client/server to generate the image/sound/input data)

**Reply by Mr-LZG** [reply]
I'm using 3.x. There are several problems with using session raw data. One is that the amount of data is too large, and the other is that session raw data playback cannot jump like a video. 
How can I use the capture plug-in in 3.x, or tell me where this part of the code is in 2.x, and whether I can merge the code and use it in 3.x

**Reply by akallabeth** [reply]
you can not.
we´ve changed the proxy in 3.x to not decode any data due to performance issues - 2.x did just that, decode and reencode on proxy.

**Reply by Mr-LZG** [reply]
@akallabeth 
I have a few questions. Can you give me some advice?
1. In version 2.x, in which file and function is the code for decoding data in proxy located?
2. If the original data is used to play back the RDP session, can the jump playback function be implemented?
3. If raw data is used to play back the RDP session, is there any way to reduce the amount of data (which can reduce the requirements for playback quality)?

**Reply by Mr-LZG** [reply]
I have a suggestion, could you consider decompressing the gfx data when the capture plug-in is enabled?

**Reply by akallabeth** [reply]
@Mr-LZG you can do whatever you deem fit for your purpose, but no, this is not worth the effort for us.

1. you can already dump raw session data
2. you can always recreate the session from that.
3. it does not cost a lot of extra CPU/bandwidth to create such a dump
4. if you so desperately need a capture, write a decoder and let it run in parallel

what we should (and will) do is remove the capture plugin as it is useless with FreeRDP 3 though.

**Reply by Mr-LZG** [reply]
I noticed that there are some settings in the freerdp code. Can you tell me what these settings do, or where can I get these instructions? 
1. FreeRDP_TransportDump
2. FreeRDP_TransportDumpFile
3. FreeRDP_TransportDumpReplay
4. FreeRDP_DumpRemoteFxFile
5. FreeRDP_PlayRemoteFxFile

**Reply by akallabeth** [reply]
1. `FreeRDP_TransportDump` enables dumping of session data to a file
2. `FreeRDP_TransportDumpReplay` enables crude replay of a recorded session from a file
3. `FreeRDP_TransportDumpReplay` the file to use for the settings above
4. `FreeRDP_DumpRemoteFxFile` and `FreeRDP_PlayRemoteFxFile` are old test settings to dump windows 7 RFX commands to a file or replay one.
